nff/io/ase_calcs.py

```diff
@@ -197,15 +210,27 @@ def calculate(
             self.results["embedding"] = embedding
 
         if requires_stress:
-            stress = prediction["stress_volume"].detach().cpu().numpy() * (
-                1 / const.EV_TO_KCAL_MOL
-                # TODO change to more general prediction
-            )
-            self.results["stress"] = stress * (1 / atoms.get_volume())
+            if isinstance(self.model, NffScaleMACE):#the implementation of stress calculation in MACE is a bit different and hence this is needed (ASE_suit: mace/mace/calculators/mace.py)
+               # print(f"Original stress shape: {prediction['stress'].shape}")
+                if prediction["stress"].ndim==1:
+                    try:
+                        prediction["stress"] = prediction["stress"].reshape(3, 3)
+                       # print("Reshaped stress to 3*3:", prediction["stress"])
+                    except ValueError as e:
+                        raise ValueError(f"Error reshaping stress tensor: {e}, shape: {prediction['stress'].shape}")
+
+               # print("The current dimensions of the stress tensor is:", prediction["stress"].ndim)
+                
+                self.results["stress"] = full_3x3_to_voigt_6_stress(
+                    torch.mean(prediction["stress"], dim=0).cpu().numpy()*self.energy_units_to_eV/1.0)#1.0 as volume is in A**3, but conversion for energy is needed
```

Comment:
I am not sure I understand why you are dividing by 1.0? You can comment that you are converting to eV/&angst;<sup>3</sup> and I don't think further explanation is needed beyond that and the 1.0 is redundant.